class-srcset-callback.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+class WP_Ricg_Content_Filter {
+	private $attachments_loop;
+	private $attachments_array;
+	
+	function __construct( $attachments_loop ) {
+		$this->attachments_loop = $attachments_loop;
+		$upload_dir = wp_upload_dir();
+		$upload_dir = $upload_dir['baseurl'] . '/';
+		
+		while( $attachments_loop->have_posts() ) :
+			$attachments_loop->the_post();
+			
+			// Get the image meta data.
+			if ( is_array( $img_meta = wp_get_attachment_metadata( get_the_ID() ) ) ) {
+				
+				// Get base URL.
+				$base_url = $upload_dir . pathinfo( $img_meta['file'], PATHINFO_DIRNAME );
+				$base_url = untrailingslashit( $base_url ) . '/';
+				
+				// Add default/full size image.
+				$this->attachments_array[ $upload_dir . $img_meta['file'] ] = get_the_ID();
+				
+				// Build from sizes array.
+				$img_sizes = $img_meta['sizes'];
+
+				foreach( $img_sizes as $img_size ) {
+					$this->attachments_array[ $base_url . $img_size['file'] ] = get_the_ID();
+				}
+			}
+		endwhile;
+
+		wp_reset_postdata();
+	}
+	
+	/**
+	 * Callback function for tevkori_filter_content_images.
+	 *
+	 * @since 3.0
+	 *
+	 * @see tevkori_filter_content_images
+	 * @param array $matches Array containing the regular expression matches.
+	 */
+	public function callback( $matches ) {
+		$atts = $matches[1];
+		$sizes = $srcset = '';
+	
+		// Check if srcset attribute is not already present.
+		if ( false !== strpos( 'srcset="', $atts ) ) {
+
+			// Get the url of the original image.
+			preg_match( '/src="(.+?)(\-([0-9]+)x([0-9]+))?(\.[a-zA-Z]{3,4})"/i', $atts, $url_matches );
```

Comment:
The regex won't match JPEG 2000 (.JP2).
Probably better to use ```\.\w{3,4}``` instead of ```\.[a-zA-Z]{3,4}``` but have to make sure that it only matches if it's at the end to avoid problems with URLs that have an IP address as host.